x-pack/plugins/security_solution/public/common/store/data_table/middlware_local_storage.test.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../mock/match_media';
+import {
+  mockGlobalState,
+  SUB_PLUGINS_REDUCER,
+  defaultHeaders,
+  createSecuritySolutionStorageMock,
+  kibanaObservable,
+} from '../../mock';
+
+import type { State } from '..';
+import { createStore } from '..';
+
+import { addTableInStorage } from '../../../timelines/containers/local_storage';
+import { Direction } from '../../../../common/search_strategy';
+import { TableId, dataTableActions } from '@kbn/securitysolution-data-table';
+
+const {
+  applyDeltaToColumnWidth,
+  removeColumn,
+  updateColumnOrder,
+  updateColumns,
+  updateColumnWidth,
+  updateItemsPerPage,
+  updateSort,
+  upsertColumn,
+} = dataTableActions;
+
+jest.mock('../../../timelines/containers/local_storage');
+
+const addTableInStorageMock = addTableInStorage as jest.Mock;
+
+describe('epicLocalStorage', () => {
```

Comment:
These tests used to use `shallow` rendering, which wasn't actually required. It's possible to test the middleware without the React "context" around it. This speeds up these tests quite a bit.

Also, no fake timers required anymore, yay 🎉